### PR TITLE
add setup for Python type checking

### PR DESCRIPTION
**Pull request by dfrankland**

### CODE REVIEW

File: context/src/env/validator.rs [hunk 1]
```diff
@@ -277,7 +277,7 @@ pub struct EnvValidationFlatIssue {
     pub error_message: String,
 }
 
-#[cfg_attr(feature = "pyo3", pymethods)]
+#[cfg_attr(feature = "pyo3", gen_stub_pymethods, pymethods)]
```
Comment (dfrankland) [inline context/src/env/validator.rs:280]: a heads up to anyone reading this: order matters with `gen_stub_pymethods`—it needs to be before `pymethods` to generate types correctly. It's not documented and can cause some surprising results.

File: context-py/tests/test_parse_and_validate.py [hunk 2]
```diff
@@ -12,6 +12,7 @@ def test_env_parse_and_validate():
     }
 
     ci_info = env_parse(env_vars)
+    assert ci_info is not None
```
Comment (dfrankland) [inline context-py/tests/test_parse_and_validate.py:15]: Caught by `pyright` 🤙

File: context-py/tests/test_parse_and_validate.py [hunk 3]
```diff
@@ -256,7 +257,7 @@ def test_repo_validate():
 
     from context_py import BundleRepo, RepoUrlParts, RepoValidationLevel, repo_validate
 
-    repo = RepoUrlParts("github", "trunk-io", "analytics-cli")
+    repo = RepoUrlParts(host="github", owner="trunk-io", name="analytics-cli")
```
Comment (dfrankland) [inline context-py/tests/test_parse_and_validate.py:260]: easier to understand 👍

File: context-py/README.md
Comment (dfrankland) [inline context-py/README.md:1]: Now that there's a `requirements` file in place it's much easier to develop locally 😄

File: .github/actions/setup_build_wasm/action.yaml [hunk 4]
```diff
@@ -4,9 +4,8 @@ description: Setup and build wasm
 runs:
   using: composite
   steps:
-    - uses: dtolnay/rust-toolchain@nightly
-      with:
-        targets: wasm32-unknown-unknown
+    - name: Setup Rust & Cargo
+      uses: ./.github/actions/setup_rust_cargo
```
Comment (dfrankland) [inline .github/actions/setup_build_wasm/action.yaml:8]: I moved this for consistency and hopefully better caching 🤷

File: .github/workflows/pull_request.yml [hunk 5]
```diff
@@ -103,7 +103,8 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - uses: actions-rust-lang/setup-rust-toolchain@v1
+      - name: Setup and build pyo3
+        uses: ./.github/actions/setup_build_pyo3
```
Comment (dfrankland) [inline .github/workflows/pull_request.yml:107]: Python stubs and bindings must be built prior to running checks

**Review by TylerJang27 — APPROVED** [review]
Thanks for the cleanup 🙏

**Pull request merged**